Blinka Install Errors and Warnings

### Board Name

Raspberry Pi 5

### Steps

1. Follow directions at https://learn.adafruit.com/circuitpython-on-raspberrypi-linux/installing-circuitpython-on-raspberry-pi

### Description

Summary of errors include:
- DEPRECATION: Building 'sysv_ipc' using the legacy setup.py bdist_wheel mechanism
- error: externally-managed-environment (related to swig, pre-req for lgpio)
- HTTP redirects when downloading lgpio.zip
- Errors building lgpio
- SetuptoolsDeprecationWarning: setup.py install is deprecated (lgpio related)
- EasyInstallDeprecationWarning: easy_install command is deprecated (lgpio related)

Some of these are repeated.

### Additional information

Here is most of the output with the error messages:
```
Blinka DEPRECATION: Building 'sysv_ipc' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'sysv_ipc'. Discussion can be found at https://github.com/pypa/pip/issues/6334
Blinka Building wheel for sysv_ipc (setup.py): started
Blinka Building wheel for sysv_ipc (setup.py): finished with status 'done'
Blinka Created wheel for sysv_ipc: filename=sysv_ipc-1.1.0-cp313-cp313-linux_aarch64.whl size=66928 sha256=2417600cb58c464d9ab559bef870fe43290854a422fc8a171ad20131107f7a46
Blinka Stored in directory: /home/pi/.cache/pip/wheels/0d/1a/84/e928fcee2420c31e26381d3e5ebcc6fdb65f1c9221cfb8c9ca
Blinka Successfully built sysv_ipc
Blinka Installing collected packages: sysv_ipc, pyusb, binho-host-adapter, Adafruit-PureIO, pyftdi, Adafruit-Circuitpython-ConnectionManager, adafruit-circuitpython-requests, adafruit-circuitpython-busdevice, adafruit-circuitpython-typing, adafruit-blinka
Blinka
Blinka Successfully installed Adafruit-Circuitpython-ConnectionManager-3.1.5 Adafruit-PureIO-1.1.11 adafruit-blinka-8.66.2 adafruit-circuitpython-busdevice-5.2.13 adafruit-circuitpython-requests-4.1.13 adafruit-circuitpython-typing-1.12.2 binho-host-adapter-0.1.6 pyftdi-0.57.1 pyusb-1.3.1 sysv_ipc-1.1.0
Adding user to the group: i2c.
Adding user to the group: spi.
Adding user to the group: gpio.
Adding user to the group: input.
Adding user to the group: video.
Detected Raspberry Pi 5, applying additional fixes...
Blinka Reading package lists...
Blinka
Blinka Building dependency tree...
Blinka
Blinka Reading state information...
Blinka
Blinka wget is already the newest version (1.25.0-2).
wget set to manually installed.
python3-dev is already the newest version (3.13.5-1).
python3-dev set to manually installed.
python3-setuptools is already the newest version (78.1.1-0.1).
Suggested packages:
Blinka swig-doc swig-examples
Blinka The following NEW packages will be installed:
Blinka swig
Blinka 0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 1,406 kB of archives.
After this operation, 6,122 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian trixie/main arm64 swig arm64 4.3.0-1 [1,406 kB]
Blinka Fetched 1,406 kB in 0s (6,245 kB/s)
Blinka Selecting previously unselected package swig.
(Reading database ...
Blinka (Reading database ... 5%
Blinka (Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
Blinka (Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
Blinka (Reading database ... 45%
(Reading database ... 50%
Blinka (Reading database ... 55%
Blinka
Blinka (Reading database ...
Blinka 60%
Blinka (Reading database ... 65%
(Reading database ... 70%
Blinka (Reading database ...
Blinka 75%
Blinka (Reading database ...
Blinka 80%
Blinka (Reading database ... 85%
Blinka (Reading database ... 90%
Blinka
Blinka (Reading database ...
Blinka 95%
Blinka (Reading database ...
Blinka 100%
(Reading database ... 119867 files and directories currently installed.)
Blinka Preparing to unpack .../swig_4.3.0-1_arm64.deb ...
Blinka Unpacking swig (4.3.0-1) ...
Blinka Setting up swig (4.3.0-1) ...
Blinka Processing triggers for man-db (2.13.1-1) ...
Blinka error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.

    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
Blinka --2025-10-11 13:42:57--  https://github.com/adafruit/Raspberry-Pi-Installer-Scripts/raw/refs/heads/main/packages/lgpio.zip
Blinka Resolving github.com (github.com)...
Blinka 140.82.114.3
Blinka Connecting to github.com (github.com)|140.82.114.3|:443...
Blinka connected.
Blinka HTTP request sent, awaiting response...
Blinka 302 Found
Blinka Location: https://raw.githubusercontent.com/adafruit/Raspberry-Pi-Installer-Scripts/refs/heads/main/packages/lgpio.zip [following]
--2025-10-11 13:42:57--  https://raw.githubusercontent.com/adafruit/Raspberry-Pi-Installer-Scripts/refs/heads/main/packages/lgpio.zip
Blinka Resolving raw.githubusercontent.com (raw.githubusercontent.com)...
Blinka 185.199.111.133
Blinka , 185.199.108.133, 185.199.109.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443...
Blinka connected.
Blinka HTTP request sent, awaiting response...
Blinka 200 OK
Blinka Length:
Blinka 2739152 (2.6M) [application/zip]
Blinka Saving to: ‘lgpio.zip’
Blinka 0K .
Blinka ....
Blinka .
Blinka ...
Blinka .
Blinka ...
Blinka ..
Blinka ..... ..
Blinka ........ .
Blinka .........
Blinka .......
Blinka ...  1% 2.69M 1s
    50K ......
Blinka .... ...
Blinka ....... ..
Blinka .......
Blinka . .......... .
Blinka ....
Blinka .....
Blinka 3% 8.04M 1s
   100K .......... .
Blinka ......... .
Blinka ......
Blinka ..
Blinka . .......... ..........  5% 3.82M 1s
   150K .........
Blinka . .......... .
Blinka ....
Blinka ...
Blinka .. .......... .
Blinka .........  7% 11.3M 1s
   200K ....
Blinka ..
Blinka .... ....
Blinka ...... ..
Blinka ...
Blinka ..... .......... ..........  9% 20.5M 0s
   250K ....
Blinka ...
Blinka ... ..........
Blinka ......
Blinka .... ......
Blinka .... ........
Blinka .. 11% 4.60M 0s
   300K .......... ........
Blinka .. .......... ...
Blinka .
Blinka ...... .........
Blinka . 13% 40.9M 0s
Blinka 350K .......... ...
Blinka ....... .......... ..
Blinka ......
Blinka .. ........
Blinka ..
Blinka 14% 8.22M 0s
   400K .......... ....
Blinka ...... ......
Blinka .... .......... ......
Blinka .... 16% 44.2M 0s
   450K .
Blinka ......... ..
Blinka .......
Blinka . ........
Blinka .. ....
Blinka ...... .......... 18% 32.9M 0s
   500K ..........
Blinka ....
Blinka ...... ......
Blinka .... .
Blinka ......... ..
Blinka .....
Blinka ... 20% 29.0M 0s
   550K .......... .......... ....
Blinka ...... ........
Blinka ..
Blinka .......
Blinka ... 22% 6.29M 0s
   600K ......
Blinka .... .........
Blinka . .......... ........
Blinka ..
Blinka .......... 24% 44.3M 0s
   650K ....
Blinka .....
Blinka . ..........
Blinka ........
Blinka .. .......... .......... 26% 34.1M 0s
   700K ..
Blinka ........
Blinka ........
Blinka ..
Blinka .......... ....
Blinka ...... .....
Blinka ..... 28% 16.9M 0s
Blinka 750K .........
Blinka . ......
Blinka .... .......
Blinka ... .......... ........
Blinka .. 29%
Blinka 27.9M 0s
   800K .......... ....
Blinka ...... ........
Blinka ..
Blinka ..........
Blinka .......... 31% 35.5M 0s
   850K .......... ..
Blinka ........ .....
Blinka ...
Blinka .. .......... .
Blinka ...
Blinka .....
Blinka . 33% 29.1M 0s
   900K .......... .......... ......
Blinka .... .
Blinka ......... ..
Blinka ........
Blinka 35% 42.0M 0s
   950K ........
Blinka .. ..........
Blinka ....
Blinka ...... .....
Blinka .....
Blinka ..........
Blinka 37% 28.9M 0s
  1000K ......
Blinka .... .
Blinka ......... ..
Blinka ........ .
Blinka .......
Blinka .. ..........
Blinka 39% 33.5M 0s
  1050K .......... ..........
Blinka ..........
Blinka ......
Blinka .... .
Blinka ......... 41% 44.2M 0s
  1100K ..
Blinka ...
Blinka ..... .......... .......... ....
Blinka ...... .........
Blinka . 42% 41.6M 0s
Blinka 1150K ........
Blinka .. ......
Blinka .... .......
Blinka ... .......... ........
Blinka .. 44% 28.9M 0s
  1200K ......
Blinka .... ....
Blinka ...
Blinka ... .......... .......... ......
Blinka .... 46% 15.6M 0s
  1250K
Blinka .......... ..
Blinka ........ ...
Blinka .....
Blinka .. ..........
Blinka .......... 48% 26.7M 0s
  1300K ..........
Blinka .......... .....
Blinka .
Blinka .... .
Blinka ......... ..
Blinka ........ 50% 30.3M 0s
  1350K .
Blinka ......... .......... ....
Blinka ...
Blinka ... ..........
Blinka .....
Blinka ..... 52% 41.8M 0s
  1400K ......
Blinka .... .......... .
Blinka ......... ........
Blinka .. ........
Blinka .. 54% 44.0M 0s
  1450K ....
Blinka ...... ........
Blinka ..
Blinka ..........
Blinka .......... .......... 56% 7.24M 0s
  1500K ..
Blinka ...
Blinka ..... ........
Blinka .. ........
Blinka .. ....
Blinka ...... .......
Blinka ... 57% 28.9M 0s
  1550K .......... ......
Blinka .... ....
Blinka ...... ..
Blinka ........ ........
Blinka .. 59% 26.5M 0s
  1600K ....
Blinka ...... .......... ..........
Blinka .......... ....
Blinka ..
Blinka .... 61% 28.3M 0s
  1650K
Blinka .......... ..
Blinka ......
Blinka .. ........
Blinka .. ......
Blinka .... ....
Blinka ..
Blinka .... 63% 9.24M 0s
  1700K ..........
Blinka .......... .
Blinka ......... .......... ..
Blinka ..
Blinka ...... 65% 42.8M 0s
  1750K ........
Blinka .. ..........
Blinka ....
Blinka ...... ....
Blinka ...... .......... 67% 41.6M 0s
  1800K ......
Blinka .... ......
Blinka .... ..
Blinka .....
Blinka ... ........
Blinka .. ..
Blinka ........ 69% 30.0M 0s
  1850K .......... ..........
Blinka ..........
Blinka ......
Blinka .... ....
Blinka ...... 71% 41.7M 0s
  1900K ..
Blinka ....
Blinka .... .......... .......... ....
Blinka ....
Blinka .. .......... 72% 38.0M 0s
Blinka 1950K .......
Blinka ... ......
Blinka .
Blinka ... .......... ..
Blinka ........ .......
Blinka ... 74% 26.2M 0s
  2000K .......... ....
Blinka .....
Blinka . ..........
Blinka .......
Blinka ... ......
Blinka ....
Blinka 76% 30.2M 0s
  2050K .......... .......... ........
Blinka .. ..
Blinka ........ ....
Blinka ...... 78%
Blinka 30.8M 0s
  2100K ..........
Blinka ........
Blinka .. .......... .......... ..
Blinka ........ 80% 42.7M
Blinka 0s
  2150K ........
Blinka ..
Blinka .......... ....
Blinka ......
Blinka .......... .......... 82% 34.0M 0s
  2200K ......
Blinka .... .......
Blinka ... ..
Blinka ........ ...
Blinka .....
Blinka .. ......
Blinka .... 84% 20.2M 0s
  2250K .......... ..........
Blinka ..........
Blinka ......
Blinka ...
Blinka . .......... 85% 43.7M 0s
  2300K ..
Blinka ........ ......
Blinka .... .......... ....
Blinka ...... .......
Blinka ... 87% 16.5M 0s
Blinka 2350K ..
Blinka ........ ......
Blinka .... .....
Blinka ..... ..
Blinka ..
Blinka ...... ........
Blinka .. 89% 28.7M
Blinka 0s
  2400K .......... ....
Blinka ..
Blinka .... ..........
Blinka .......
Blinka ... ......
Blinka .... 91% 43.1M 0s
  2450K .......... ..
Blinka ........ .......... .......... .......... 93% 36.9M 0s
  2500K ..........
Blinka .......... ......
Blinka .... .......... ..
Blinka ........ 95% 30.3M 0s
  2550K .......... .......... ....
Blinka ...... ..
Blinka ........
Blinka ....
Blinka ...... 97% 42.6M 0s
  2600K ......
Blinka .... ........
Blinka .. ..
Blinka ........ ........
Blinka .. ....
Blinka ...... 99% 45.0M 0s
  2650K ....
Blinka ...
Blinka ... .......... ....
Blinka 100% 23.6M=0.1s
Blinka 2025-10-11 13:42:58 (17.7 MB/s) - ‘lgpio.zip’ saved [2739152/2739152]
Blinka Archive:  lgpio.zip
Blinka b959a17d723360e85648316757b02dbea9902feb
Blinka creating: lg/
Blinka inflating: lg/.gitignore
Blinka
Blinka creating: lg/CFG/
  inflating: lg/CFG/.lg_secret
Blinka creating: lg/CFG/cgi/
 extracting: lg/CFG/cgi/.git-dummy
  inflating: lg/CFG/permits
Blinka creating: lg/DOC/
   creating: lg/DOC/HTML/
Blinka linking: lg/DOC/HTML/EXAMPLES    -> ../../EXAMPLES
   creating: lg/DOC/HTML/images/
Blinka inflating: lg/DOC/HTML/images/LDR-fritz.png
Blinka
Blinka inflating: lg/DOC/HTML/images/LDR-gnup-1.png
Blinka
Blinka inflating: lg/DOC/HTML/images/LDR-gnup-2.png
Blinka
Blinka inflating: lg/DOC/HTML/images/LDR-photo.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/LDR-wave-1.png
Blinka inflating: lg/DOC/HTML/images/LDR-wave-2.png
Blinka
Blinka inflating: lg/DOC/HTML/images/LDR-wave-3.png
Blinka
Blinka inflating: lg/DOC/HTML/images/breadboard.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/caps.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/driver.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/faq-i2c-ss.png
Blinka
Blinka inflating: lg/DOC/HTML/images/faq-i2c.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/faq-serial.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/faq-spi.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/faq1.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/faq2.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/faq3.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/imu-1.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/imu-2.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/imu-3.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ir-fritz.png
Blinka
Blinka inflating: lg/DOC/HTML/images/ir-motion.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ir-photo.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ir-rx.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ir-wave-1.png
Blinka
Blinka inflating: lg/DOC/HTML/images/ir-wave-2.png
Blinka inflating: lg/DOC/HTML/images/ir-wave-3.png
Blinka
Blinka inflating: lg/DOC/HTML/images/keypad.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/lcd.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ldr-cap.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ldr.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/leds.jpg
Blinka
Blinka extracting: lg/DOC/HTML/images/lg-logo.gif
Blinka inflating: lg/DOC/HTML/images/meter.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/motor.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/msp430.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/nano.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/oled-2.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/oled.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/pins.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/pisc-1.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/pisc-2.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/pisc-3.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/pot.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/pro-mini.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/psu.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/re-fritz.png
Blinka
Blinka inflating: lg/DOC/HTML/images/re-photo.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/re-wave-1.png
Blinka inflating: lg/DOC/HTML/images/re-wave-2.png
Blinka
Blinka inflating: lg/DOC/HTML/images/remote-1.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/remote-2.jpg
Blinka inflating: lg/DOC/HTML/images/reverse.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/rf-rx-2.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/rf-rx.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/rf-tx.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/rotary.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/rpi.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/serial.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/servo.jpg
Blinka
Blinka extracting: lg/DOC/HTML/images/sidebar.gif
Blinka inflating: lg/DOC/HTML/images/son-fritz.png
Blinka
Blinka inflating: lg/DOC/HTML/images/son-gnup-1.png
Blinka
Blinka inflating: lg/DOC/HTML/images/son-gnup-2.png
Blinka
Blinka inflating: lg/DOC/HTML/images/son-photo.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/son-wave-1.png
Blinka
Blinka inflating: lg/DOC/HTML/images/son-wave-2.png
Blinka
Blinka inflating: lg/DOC/HTML/images/son-wave-3.png
Blinka inflating: lg/DOC/HTML/images/son-wave-4.png
Blinka
Blinka inflating: lg/DOC/HTML/images/speaker.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/spi-lnx-pi3b.png
Blinka
Blinka inflating: lg/DOC/HTML/images/spi-lnx-pibr1.png
Blinka
Blinka inflating: lg/DOC/HTML/images/spi-pig-pi3b.png
Blinka
Blinka inflating: lg/DOC/HTML/images/spi-pig-pibr1.png
Blinka
Blinka inflating: lg/DOC/HTML/images/srf02.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/srf04.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/stepper.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/switches.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/topbar.gif
Blinka inflating: lg/DOC/HTML/images/transistors.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/ubec-2.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/uln2003a.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/wheel.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/wires.jpg
Blinka
Blinka inflating: lg/DOC/HTML/images/yl-40.jpg
Blinka
Blinka creating: lg/DOC/HTML/scripts/
Blinka inflating: lg/DOC/HTML/scripts/index.css
Blinka inflating: lg/DOC/HTML/scripts/standard.css
Blinka inflating: lg/DOC/README
   creating: lg/DOC/bin/
Blinka inflating: lg/DOC/bin/backup.sh
  inflating: lg/DOC/bin/body.py
Blinka inflating: lg/DOC/bin/build_site.py
  inflating: lg/DOC/bin/cmakdoc.py
Blinka
Blinka inflating: lg/DOC/bin/dmakdoc.py
Blinka inflating: lg/DOC/bin/examples.py
Blinka
Blinka inflating: lg/DOC/bin/html.py
Blinka
Blinka inflating: lg/DOC/bin/purge.sh
  inflating: lg/DOC/bin/pymakdoc.py
Blinka
Blinka inflating: lg/DOC/bin/smakdoc.py
Blinka
Blinka inflating: lg/DOC/bin/tidy.py
Blinka inflating: lg/DOC/bin/updatesql.py
Blinka inflating: lg/DOC/cdoc
Blinka creating: lg/DOC/dbase/
  inflating: lg/DOC/dbase/lg.sqlite
Blinka
Blinka inflating: lg/DOC/hdoc
Blinka inflating: lg/DOC/makedoc
Blinka inflating: lg/DOC/pdoc
   creating: lg/DOC/src/
Blinka creating: lg/DOC/src/defs/
  inflating: lg/DOC/src/defs/download.def
Blinka inflating: lg/DOC/src/defs/examples.def
Blinka
Blinka inflating: lg/DOC/src/defs/faq.def
  inflating: lg/DOC/src/defs/index.def
Blinka inflating: lg/DOC/src/defs/permits.def
Blinka
Blinka inflating: lg/DOC/src/defs/rgpiod.def
Blinka
Blinka inflating: lg/DOC/src/defs/rgs.def
Blinka
Blinka inflating: lg/DOC/src/defs/scripts.def
Blinka
Blinka creating: lg/EXAMPLES/
   creating: lg/EXAMPLES/lgpio/
Blinka inflating: lg/EXAMPLES/lgpio/28BYJ_48.c
Blinka inflating: lg/EXAMPLES/lgpio/NRF24.c
Blinka
Blinka inflating: lg/EXAMPLES/lgpio/bench.c
Blinka inflating: lg/EXAMPLES/lgpio/chipline.c
Blinka inflating: lg/EXAMPLES/lgpio/dhtxx.c
Blinka
Blinka inflating: lg/EXAMPLES/lgpio/lg_ads1x15.h
Blinka inflating: lg/EXAMPLES/lgpio/lg_mcp3008.h
Blinka inflating: lg/EXAMPLES/lgpio/lg_mcp3202.h
Blinka inflating: lg/EXAMPLES/lgpio/lg_mcp4131.h
Blinka inflating: lg/EXAMPLES/lgpio/lgl_ads1x15.c
Blinka
Blinka inflating: lg/EXAMPLES/lgpio/lgl_mcp3008.c
Blinka
Blinka inflating: lg/EXAMPLES/lgpio/lgl_mcp3202.c
Blinka inflating: lg/EXAMPLES/lgpio/lgl_mcp4131.c
Blinka inflating: lg/EXAMPLES/lgpio/monitor.c
Blinka inflating: lg/EXAMPLES/lgpio/rotary_encoder.c
Blinka
Blinka inflating: lg/EXAMPLES/lgpio/sonar_ranger.c
Blinka inflating: lg/EXAMPLES/lgpio/tx_pulse.c
Blinka inflating: lg/EXAMPLES/lgpio/tx_servo.c
Blinka inflating: lg/EXAMPLES/lgpio/tx_wave.c
Blinka creating: lg/EXAMPLES/py_lgpio/
  inflating: lg/EXAMPLES/py_lgpio/28BYJ_48.py
Blinka inflating: lg/EXAMPLES/py_lgpio/DHT.py
Blinka inflating: lg/EXAMPLES/py_lgpio/NRF24.py
Blinka
Blinka inflating: lg/EXAMPLES/py_lgpio/bench.py
Blinka inflating: lg/EXAMPLES/py_lgpio/chipline.py
  inflating: lg/EXAMPLES/py_lgpio/errors.py
Blinka
Blinka inflating: lg/EXAMPLES/py_lgpio/lg_ads1x15.py
Blinka
Blinka inflating: lg/EXAMPLES/py_lgpio/lg_mcp3008.py
Blinka inflating: lg/EXAMPLES/py_lgpio/lg_mcp3202.py
Blinka inflating: lg/EXAMPLES/py_lgpio/lg_mcp4131.py
Blinka inflating: lg/EXAMPLES/py_lgpio/lg_mcp4251.py
Blinka
Blinka inflating: lg/EXAMPLES/py_lgpio/monitor.py
Blinka inflating: lg/EXAMPLES/py_lgpio/motor_sync.py
Blinka
Blinka inflating: lg/EXAMPLES/py_lgpio/rotary_encoder.py
Blinka inflating: lg/EXAMPLES/py_lgpio/sonar_ranger.py
Blinka
Blinka inflating: lg/EXAMPLES/py_lgpio/testbed.py
Blinka inflating: lg/EXAMPLES/py_lgpio/tx_pulse.py
Blinka inflating: lg/EXAMPLES/py_lgpio/tx_wave.py
Blinka creating: lg/EXAMPLES/py_rgpio/
  inflating: lg/EXAMPLES/py_rgpio/28BYJ_48.py
Blinka inflating: lg/EXAMPLES/py_rgpio/DHT.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/DS18B20.py
Blinka inflating: lg/EXAMPLES/py_rgpio/NRF24.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/bench.py
Blinka inflating: lg/EXAMPLES/py_rgpio/chipline.py
Blinka inflating: lg/EXAMPLES/py_rgpio/errors.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/files.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/lg_ads1x15.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/lg_mcp3008.py
Blinka inflating: lg/EXAMPLES/py_rgpio/lg_mcp3202.py
Blinka inflating: lg/EXAMPLES/py_rgpio/lg_mcp4131.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/lg_mcp4251.py
Blinka inflating: lg/EXAMPLES/py_rgpio/monitor.py
Blinka inflating: lg/EXAMPLES/py_rgpio/motor_sync.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/rotary_encoder.py
Blinka inflating: lg/EXAMPLES/py_rgpio/sonar_ranger.py
Blinka
Blinka inflating: lg/EXAMPLES/py_rgpio/testbed.py
Blinka inflating: lg/EXAMPLES/py_rgpio/tx_pulse.py
Blinka inflating: lg/EXAMPLES/py_rgpio/tx_wave.py
Blinka creating: lg/EXAMPLES/rgpio/
  inflating: lg/EXAMPLES/rgpio/28BYJ_48.c
Blinka inflating: lg/EXAMPLES/rgpio/DS18B20.c
Blinka inflating: lg/EXAMPLES/rgpio/NRF24.c
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/bench.c
Blinka inflating: lg/EXAMPLES/rgpio/chipline.c
Blinka inflating: lg/EXAMPLES/rgpio/dhtxxd.c
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/errors.c
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/files.c
Blinka inflating: lg/EXAMPLES/rgpio/lg_ads1x15.h
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/lg_mcp3008.h
Blinka inflating: lg/EXAMPLES/rgpio/lg_mcp3202.h
Blinka inflating: lg/EXAMPLES/rgpio/lg_mcp4131.h
  inflating: lg/EXAMPLES/rgpio/lg_mcp4251.h
Blinka inflating: lg/EXAMPLES/rgpio/lgr_ads1x15.c
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/lgr_mcp3008.c
Blinka inflating: lg/EXAMPLES/rgpio/lgr_mcp3202.c
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/lgr_mcp4131.c
Blinka inflating: lg/EXAMPLES/rgpio/lgr_mcp4251.c
Blinka inflating: lg/EXAMPLES/rgpio/monitor.c
Blinka
Blinka inflating: lg/EXAMPLES/rgpio/rotary_encoder.c
Blinka inflating: lg/EXAMPLES/rgpio/sonar_ranger.c
Blinka inflating: lg/EXAMPLES/rgpio/tx_pulse.c
Blinka inflating: lg/EXAMPLES/rgpio/tx_wave.c
   creating: lg/EXAMPLES/rgs/
Blinka inflating: lg/EXAMPLES/rgs/chipline.rgs
  inflating: lg/Makefile
Blinka creating: lg/PY_LGPIO/
Blinka inflating: lg/PY_LGPIO/LICENSE
Blinka inflating: lg/PY_LGPIO/README.md
Blinka inflating: lg/PY_LGPIO/lgpio.i
Blinka
Blinka inflating: lg/PY_LGPIO/lgpio_extra.py
Blinka
Blinka inflating: lg/PY_LGPIO/setup.py
Blinka creating: lg/PY_RGPIO/
Blinka inflating: lg/PY_RGPIO/LICENSE
Blinka inflating: lg/PY_RGPIO/README.md
  inflating: lg/PY_RGPIO/rgpio.py
Blinka
Blinka inflating: lg/PY_RGPIO/setup.py
Blinka inflating: lg/README
Blinka inflating: lg/README.md
Blinka inflating: lg/UNLICENCE
Blinka inflating: lg/lgCfg.c
Blinka
Blinka inflating: lg/lgCfg.h
Blinka inflating: lg/lgCmd.c
Blinka
Blinka inflating: lg/lgCmd.h
Blinka inflating: lg/lgCtx.c
Blinka inflating: lg/lgCtx.h
Blinka
Blinka inflating: lg/lgDbg.c
Blinka inflating: lg/lgDbg.h
Blinka inflating: lg/lgErr.c
Blinka
Blinka inflating: lg/lgExec.c
Blinka
Blinka inflating: lg/lgFile.c
Blinka
Blinka inflating: lg/lgGpio.c
Blinka
Blinka inflating: lg/lgGpio.h
Blinka inflating: lg/lgHdl.c
Blinka inflating: lg/lgHdl.h
  inflating: lg/lgI2C.c
Blinka
Blinka inflating: lg/lgMD5.c
Blinka
Blinka inflating: lg/lgMD5.h
Blinka inflating: lg/lgNotify.c
Blinka
Blinka inflating: lg/lgPthAlerts.c
Blinka
Blinka inflating: lg/lgPthAlerts.h
Blinka inflating: lg/lgPthSocket.c
Blinka inflating: lg/lgPthTx.c
Blinka
Blinka inflating: lg/lgPthTx.h
Blinka inflating: lg/lgSPI.c
Blinka
Blinka inflating: lg/lgScript.c
Blinka
Blinka inflating: lg/lgSerial.c
Blinka inflating: lg/lgThread.c
Blinka
Blinka inflating: lg/lgUtil.c
Blinka
Blinka inflating: lg/lgpio.3
Blinka
Blinka inflating: lg/lgpio.h
Blinka
Blinka inflating: lg/rgpio.3
Blinka
Blinka inflating: lg/rgpio.c
Blinka
Blinka inflating: lg/rgpio.h
Blinka
Blinka inflating: lg/rgpiod.1
Blinka
Blinka inflating: lg/rgpiod.c
Blinka
Blinka inflating: lg/rgpiod.h
Blinka
Blinka inflating: lg/rgs.1
Blinka
Blinka inflating: lg/rgs.c
Blinka
Blinka extracting: lg/v0.2.2.0
finishing deferred symbolic links:
Blinka lg/DOC/HTML/EXAMPLES   -> ../../EXAMPLES
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgCtx.o lgCtx.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgDbg.o lgDbg.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgErr.o lgErr.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgGpio.o lgGpio.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgHdl.o lgHdl.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgI2C.o lgI2C.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgNotify.o lgNotify.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgPthAlerts.o lgPthAlerts.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgPthTx.o lgPthTx.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgSerial.o lgSerial.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgSPI.o lgSPI.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgThread.o lgThread.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgUtil.o lgUtil.c
Blinka gcc -shared -pthread  -Wl,-soname,liblgpio.so.1 -o liblgpio.so.1 lgCtx.o lgDbg.o lgErr.o lgGpio.o lgHdl.o lgI2C.o lgNotify.o lgPthAlerts.o lgPthTx.o lgSerial.o lgSPI.o lgThread.o lgUtil.o
Blinka ln -fs liblgpio.so.1 liblgpio.so
Blinka strip --strip-unneeded liblgpio.so
Blinka size     liblgpio.so
Blinka text	   data	    bss	    dec	    hex	filename
  80622	   3412	 111152	 195186	  2fa72	liblgpio.so
Blinka gcc -O3 -Wall -pthread -fpic  -c -o rgpio.o rgpio.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgCfg.o lgCfg.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgMD5.o lgMD5.c
Blinka gcc -shared -pthread  -Wl,-soname,librgpio.so.1 -o librgpio.so.1 rgpio.o lgCfg.o lgErr.o lgDbg.o lgMD5.o
Blinka ln -fs librgpio.so.1 librgpio.so
Blinka strip --strip-unneeded librgpio.so
Blinka size     librgpio.so
Blinka text	   data	    bss	    dec	    hex	filename
  44499	   2848	   6160	  53507	   d103	librgpio.so
Blinka gcc -O3 -Wall -pthread -fpic    -c -o rgpiod.o rgpiod.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgCmd.o lgCmd.c
Blinka lgCmd.c: In function ‘cmdParse’:
Blinka lgCmd.c:554:22: warning: ‘var’ may be used uninitialized [-Wmaybe-uninitialized]
  554 |                tI[0] = var;
      |                ~~~~~~^~~~~
Blinka lgCmd.c:482:14: note: ‘var’ declared here
  482 |    uintmax_t var;
      |              ^~~
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgExec.o lgExec.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgFile.o lgFile.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgPthSocket.o lgPthSocket.c
Blinka gcc -O3 -Wall -pthread -fpic    -c -o lgScript.o lgScript.c
Blinka gcc  -o rgpiod rgpiod.o lgCfg.o lgCmd.o lgExec.o lgFile.o lgMD5.o lgPthSocket.o lgScript.o  -L. -llgpio -pthread -lrt
Blinka strip rgpiod
Blinka gcc -O3 -Wall -pthread -fpic    -c -o rgs.o rgs.c
Blinka gcc  -o rgs rgs.o lgCmd.o lgCfg.o lgDbg.o lgErr.o lgMD5.o
Blinka strip rgs
Blinka >rgs
Blinka >rgpiod
Blinka >lgpio.h
Blinka /home/pi/lg/DOC/bin/cmakdoc.py:16: SyntaxWarning: invalid escape sequence '\&'
  \&./prog
Blinka /home/pi/lg/DOC/bin/cmakdoc.py:31: SyntaxWarning: invalid escape sequence '\&'
  \&./prog
Blinka >rgpio.h
Blinka /home/pi/lg/DOC/bin/cmakdoc.py:16: SyntaxWarning: invalid escape sequence '\&'
  \&./prog
Blinka /home/pi/lg/DOC/bin/cmakdoc.py:31: SyntaxWarning: invalid escape sequence '\&'
  \&./prog
Blinka touch DOC/.docs
Blinka install -m 0644 lgpio.h                  /usr/local/include
Blinka install -m 0644 rgpio.h                  /usr/local/include
Blinka install -m 0755 liblgpio.so.1 /usr/local/lib
Blinka install -m 0755 librgpio.so.1 /usr/local/lib
Blinka install -m 0755 rgpiod                   /usr/local/bin
Blinka install -m 0755 rgs                      /usr/local/bin
Blinka install -m 0644 rgpiod.1                 /usr/local/man/man1
Blinka install -m 0644 rgs.1                    /usr/local/man/man1
Blinka install -m 0644 lgpio.3                  /usr/local/man/man3
Blinka install -m 0644 rgpio.3                  /usr/local/man/man3
Blinka ldconfig
Blinka /usr/bin/python3
Blinka /usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py:90: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
Blinka /usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py:90: EasyInstallDeprecationWarning: easy_install command is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` and ``easy_install``.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://github.com/pypa/setuptools/issues/917 for details.
        ********************************************************************************

!!
  self.initialize_options()
Blinka /usr/bin/swig
Blinka /usr/bin/swig
Blinka /usr/bin/swig
Blinka /usr/bin/python3
Blinka running build_ext
Blinka building '_lgpio' extension
Blinka creating build/temp.linux-aarch64-cpython-313
Blinka aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/usr/include/python3.13 -c lgpio_wrap.c -o build/temp.linux-aarch64-cpython-313/lgpio_wrap.o
Blinka creating build/lib.linux-aarch64-cpython-313
Blinka aarch64-linux-gnu-gcc -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 build/temp.linux-aarch64-cpython-313/lgpio_wrap.o -L/usr/lib/aarch64-linux-gnu -llgpio -o build/lib.linux-aarch64-cpython-313/_lgpio.cpython-313-aarch64-linux-gnu.so
Blinka /usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py:90: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
Blinka /usr/lib/python3/dist-packages/setuptools/_distutils/cmd.py:90: EasyInstallDeprecationWarning: easy_install command is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` and ``easy_install``.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://github.com/pypa/setuptools/issues/917 for details.
        ********************************************************************************

!!
  self.initialize_options()
Blinka error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.

    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
DONE.

Settings take effect on next boot.

REBOOT NOW? [Y/n] n
Exiting without reboot.
(env) pi@raspberrypi:~ $ pip install wheel
Requirement already satisfied: wheel in /usr/lib/python3/dist-packages (0.46.1)
Requirement already satisfied: packaging>=24.0 in /usr/lib/python3/dist-packages (from wheel) (25.0)
```

I was getting a similar error with gpio and lgpio (attempted to install lgpio separate from blinka) failing to build the wheel (pyproject.toml). (See below for error log.)

I was able to install blinka successfully after using: "sudo apt install python3-pip" instead of the python-managed pip.

This is my first time contributing to a bug report, I hope it is helpful!

Edit: oops, still getting an error for rpi-lgpio! (not included here)

<details><summary>Output log with error messages:</summary>
<p>

pip3 install Adafruit-Blinka
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting Adafruit-Blinka
  Downloading https://www.piwheels.org/simple/adafruit-blinka/adafruit_blinka-9.0.4-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 2.1 MB/s eta 0:00:00
Collecting Adafruit-PlatformDetect>=3.70.1 (from Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/adafruit-platformdetect/adafruit_platformdetect-3.88.0-py3-none-any.whl (26 kB)
Collecting Adafruit-PureIO>=1.1.7 (from Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/adafruit-pureio/Adafruit_PureIO-1.1.11-py3-none-any.whl (10 kB)
Collecting binho-host-adapter>=0.1.6 (from Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/binho-host-adapter/binho_host_adapter-0.1.6-py3-none-any.whl (10 kB)
Collecting pyftdi>=0.40.0 (from Adafruit-Blinka)
  Downloading pyftdi-0.57.1-py3-none-any.whl.metadata (3.5 kB)
Collecting adafruit-circuitpython-typing (from Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/adafruit-circuitpython-typing/adafruit_circuitpython_typing-1.12.3-py3-none-any.whl (11 kB)
Collecting sysv_ipc>=1.1.0 (from Adafruit-Blinka)
  Downloading sysv_ipc-1.2.0-cp313-cp313-manylinux2014_aarch64.manylinux_2_17_aarch64.manylinux_2_28_aarch64.whl.metadata (2.4 kB)
Collecting rpi_ws281x>=4.0.0 (from Adafruit-Blinka)
  Downloading rpi_ws281x-5.0.0.tar.gz (64 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting RPi.GPIO (from Adafruit-Blinka)
  Downloading RPi.GPIO-0.7.1.tar.gz (29 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting Adafruit-Blinka-Raspberry-Pi5-Neopixel (from Adafruit-Blinka)
  Downloading Adafruit_Blinka_Raspberry_Pi5_Neopixel-1.0.0rc2-cp313-cp313-manylinux_2_24_aarch64.manylinux_2_28_aarch64.whl.metadata (547 bytes)
Collecting pyserial (from binho-host-adapter>=0.1.6->Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/pyserial/pyserial-3.5-py2.py3-none-any.whl (90 kB)
Collecting pyusb!=1.2.0,>=1.0.0 (from pyftdi>=0.40.0->Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/pyusb/pyusb-1.3.1-py3-none-any.whl (58 kB)
Collecting adafruit-circuitpython-busdevice (from adafruit-circuitpython-typing->Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/adafruit-circuitpython-busdevice/adafruit_circuitpython_busdevice-5.2.16-py3-none-any.whl (7.5 kB)
Collecting adafruit-circuitpython-requests (from adafruit-circuitpython-typing->Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/adafruit-circuitpython-requests/adafruit_circuitpython_requests-4.1.16-py3-none-any.whl (10 kB)
Collecting typing_extensions~=4.0 (from adafruit-circuitpython-typing->Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/typing-extensions/typing_extensions-4.15.0-py3-none-any.whl (44 kB)
Collecting Adafruit-Circuitpython-ConnectionManager (from adafruit-circuitpython-requests->adafruit-circuitpython-typing->Adafruit-Blinka)
  Downloading https://www.piwheels.org/simple/adafruit-circuitpython-connectionmanager/adafruit_circuitpython_connectionmanager-3.1.7-py3-none-any.whl (7.8 kB)
Downloading pyftdi-0.57.1-py3-none-any.whl (146 kB)
Downloading sysv_ipc-1.2.0-cp313-cp313-manylinux2014_aarch64.manylinux_2_17_aarch64.manylinux_2_28_aarch64.whl (72 kB)
Downloading Adafruit_Blinka_Raspberry_Pi5_Neopixel-1.0.0rc2-cp313-cp313-manylinux_2_24_aarch64.manylinux_2_28_aarch64.whl (80 kB)
Building wheels for collected packages: rpi_ws281x, RPi.GPIO
  Building wheel for rpi_ws281x (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for rpi_ws281x (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [36 lines of output]
      /tmp/pip-build-env-j9fgp793/overlay/lib/python3.13/site-packages/setuptools/dist.py:765: SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: MIT License

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      running bdist_wheel
      running build
      running build_py
      Compiling ws281x library...
      creating build/lib.linux-aarch64-cpython-313/rpi_ws281x
      copying rpi_ws281x/rpi_ws281x.py -> build/lib.linux-aarch64-cpython-313/rpi_ws281x
      copying rpi_ws281x/__init__.py -> build/lib.linux-aarch64-cpython-313/rpi_ws281x
      running build_ext
      building '_rpi_ws281x' extension
      creating build/temp.linux-aarch64-cpython-313/lib
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/dma.c -o build/temp.linux-aarch64-cpython-313/lib/dma.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/mailbox.c -o build/temp.linux-aarch64-cpython-313/lib/mailbox.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/main.c -o build/temp.linux-aarch64-cpython-313/lib/main.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/pcm.c -o build/temp.linux-aarch64-cpython-313/lib/pcm.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/pwm.c -o build/temp.linux-aarch64-cpython-313/lib/pwm.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/rpihw.c -o build/temp.linux-aarch64-cpython-313/lib/rpihw.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c lib/ws2811.c -o build/temp.linux-aarch64-cpython-313/lib/ws2811.o
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I. -I/home/pi/.env/include -I/usr/include/python3.13 -c rpi_ws281x_wrap.c -o build/temp.linux-aarch64-cpython-313/rpi_ws281x_wrap.o
      rpi_ws281x_wrap.c:154:11: fatal error: Python.h: No such file or directory
        154 | # include <Python.h>
            |           ^~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for rpi_ws281x
  Building wheel for RPi.GPIO (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for RPi.GPIO (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [37 lines of output]
      /tmp/pip-build-env-e88zd591/overlay/lib/python3.13/site-packages/setuptools/dist.py:765: SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: MIT License

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      running bdist_wheel
      running build
      running build_py
      creating build/lib.linux-aarch64-cpython-313/RPi
      copying RPi/__init__.py -> build/lib.linux-aarch64-cpython-313/RPi
      creating build/lib.linux-aarch64-cpython-313/RPi/GPIO
      copying RPi/GPIO/__init__.py -> build/lib.linux-aarch64-cpython-313/RPi/GPIO
      running build_ext
      building 'RPi._GPIO' extension
      creating build/temp.linux-aarch64-cpython-313/source
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/home/pi/.env/include -I/usr/include/python3.13 -c source/c_gpio.c -o build/temp.linux-aarch64-cpython-313/source/c_gpio.o
      source/c_gpio.c: In function ‘setup’:
      source/c_gpio.c:170:9: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
        170 |     if ((uint32_t)gpio_mem % PAGE_SIZE)
            |         ^
      source/c_gpio.c:171:34: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
        171 |         gpio_mem += PAGE_SIZE - ((uint32_t)gpio_mem % PAGE_SIZE);
            |                                  ^
      aarch64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/home/pi/.env/include -I/usr/include/python3.13 -c source/common.c -o build/temp.linux-aarch64-cpython-313/source/common.o
      source/common.c:23:10: fatal error: Python.h: No such file or directory
         23 | #include "Python.h"
            |          ^~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for RPi.GPIO
Failed to build rpi_ws281x RPi.GPIO
ERROR: Failed to build installable wheels for some pyproject.toml based projects (rpi_ws281x, RPi.GPIO)

</p>
</details>

--- a/setup.py
+++ b/setup.py
@@ -17,10 +17,6 @@ from setuptools import setup, find_packages
 here = os.path.abspath(os.path.dirname(__file__))
 
 
-def yellow_text(text: str) -> str:
-    return f"\033[33m{text}\033[0m"
-
-
 # Import the README and use it as the long-description.
 # Note: this will only work if 'README.md' is present in your MANIFEST.in file!
 with io.open(os.path.join(here, "README.rst"), encoding="utf-8") as f:
@@ -33,7 +29,6 @@ if not glob.glob("//usr//include//python3.*//Python.h"):
     )
 
 board_reqs = []
-raspberry_pi = False
 if os.path.exists("/proc/device-tree/compatible"):
     with open("/proc/device-tree/compatible", "rb") as f:
         compat = f.read()
@@ -51,11 +46,10 @@ if os.path.exists("/proc/device-tree/compatible"):
     ):
         board_reqs = [
             "rpi_ws281x>=4.0.0",
-            "lgpio;python_version<'3.13'",
+            "lgpio>=0.2.2.0",
             "RPi.GPIO",
             "Adafruit-Blinka-Raspberry-Pi5-Neopixel",
         ]
-        raspberry_pi = True
     # BeagleBone Black, Green, PocketBeagle, BeagleBone AI, etc.
     elif b"ti,am335x" in compat:
         board_reqs = ["Adafruit_BBIO"]
@@ -130,10 +124,3 @@ setup(
         "Programming Language :: Python :: Implementation :: MicroPython",
     ],
 )
-
-if raspberry_pi and os.sys.version_info >= (3, 13):
-    print(
-        yellow_text(
-            "\n*** Raspberry Pi 5 and later: lgpio will need to be installed manually. See the lgpio homepage for more details: http://abyz.me.uk/lg/download.html ***"
-        )
-    )

